task-list/config/routes.rb

```diff
@@ -3,11 +3,17 @@
   #
   root 'welcome#index'
   get 'tasks/' => 'tasks#index'
+  post 'tasks/' => 'tasks#create'
   get 'tasks/new' => 'tasks#new'
+  get 'tasks/pre_edit' => 'tasks#pre_edit'
   get 'tasks/remove' => 'tasks#remove'
   delete 'tasks/:id' => 'tasks#destroy'
   get 'tasks/:id' => 'tasks#show'
-  post 'tasks/' => 'tasks#create'
+  patch 'tasks/:id' => 'tasks#update'
+  get 'tasks/:id/edit' => 'tasks#edit'
+  patch 'tasks/:id/complete' => 'tasks#complete'
+
+  resources :tasks, :path => 'tasks'
```

Comment:
We will be talking about how to use the `resources` method in the future. For now, however, it's kind of redundant as you have all of the basic CRUD routes specified by hand already.